tensorpipe/CMakeLists.txt

```diff
@@ -134,12 +134,14 @@ endif()
 
 ## Benchmarks
 
-add_subdirectory(benchmark)
+if (TP_BUILD_BENCHMARK)
+  add_subdirectory(benchmark)
+endif()
 
 
 ## Tests
 
-if(BUILD_TESTING)
+if(TP_BUILD_TESTING)
```

Comment:
`BUILD_TESTING` is the standard CTest variable. We're currently not using CTest anymore, but if we do switch back to it down the line, we'll want to use that variable again, and it might break builds that depend on tensorpipe by `include_directory()`.

Reply:
Yeah, I saw that BUILD_TESTING is "standard", but it doesn't give us any benefit and in fact hurts downstream projects because they need to temporarily set that variable and then possibly revert it if they are using it themselves. This was the case with PyTorch (where, in fact, we weren't even resetting it correctly because we had a typo).

Reply:
Then maybe we could at least make it default to the value of `BUILD_TESTING` when `TP_BUILD_TESTING` is not set?